colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +210,29 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+async def get_cmake_version():
+    """
+    Get the version number of CMake.
+
+    The version number is parse from the output of 'cmake --version' and is
+    expected in the form 'cmake version #.#.#' where each '#' character
+    represents part of the version number. Additional text may follow, however
+    only the first line is parsed. Parsing is performed by the pkg_resources
+    package and the returned object is a Version object from that package.
+
+    :returns: The version as parsed by the pkg_resources package
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    output = await check_output([CMAKE_EXECUTABLE, '--version'])
+    lines = output.decode().splitlines()
+    ver_line = lines[0] if lines and len(lines) else None
+    if ver_line and len(ver_line):
+        # Extract just the version part of the string.
+        ver_re_str = r'^(?:cmake\s+version)?\s*(.*)$'
```

Comment:
Excellent question. I have no idea. If I don't remove the "cmake version" prefix, then the version number comes out as a LegacyVersion object, which will never pass the later checks for 3.15+. This means I have to do some parsing of the line here.

What do you think of using the following regular expression then: `r'^(?:.*)(\d+\.\d+\..*)'`. This reduces assumptions about the string. The requirements is that the pattern `#.#.<anything>` is present where `#` are numeric sequences.

Reply:
What is the problem with comparing to a `LegacyVersion`?

Reply:
Running `cmake --verison` returns a string in the form: `cmake version 3.15.0`, optionally ending in `-rc1` for release candidates (at least in English). Simply reading the line as is as giving that to `pkg_resources.parse_version()` wil always generate a `LegacyVersion` because the string isn't just a version number. As I see it, `LegacyVersion` is not comparable; that is it is equivalent to version 0. I need to drop `cmake version ` in order to generate an informative version number object. The expression I propose has fewer assumptions about the cmake version string, assuming only that it ends in a `#.#.<text>` form. This allows any prefix to be on the version number line, regardless of language.

Reply:
Note: resolving this issues will also resolve the other oustanding issues.